lib/checkbufferoverrun.cpp

```diff
@@ -1015,6 +1015,10 @@ void CheckBufferOverrun::objectIndex()
                     if (var->valueType()->pointer > obj->valueType()->pointer)
                         continue;
                 }
+                if ((obj->isCast() || (mTokenizer->isCPP() && isCPPCast(obj))) && obj->valueType()) { // allow cast to a different type
+                    if (obj->valueType()->pointer && obj->valueType()->typeSize(*mSettings) * (idx->getKnownIntValue() + 1) <= var->valueType()->typeSize(*mSettings))
```

Comment:
This should probably use `ValueFlow::isOutOfBounds` to check the bounds of `idx`, but you will need to create a `ValueFlow::Value` for the size.

Reply:
Thanks, I've tried to implement that.